File: apps/ios/MiPlancha/UsernameEmail.swift
//
//  UsernameEmail.swift
//  MiPlancha
//
//  Authentication tools.
//
//  Created by julio on 21/08/2026.
//

private let domain = "miplancha.local"

enum UsernameEmail {


    static func normalize(_ username: String) -> String {
        return username.lowercased().trimmingCharacters(in: .whitespacesAndNewlines)
    }

    static func syntheticEmail(companyCode: String, username: String) -> String {
        return "\(normalize(username))@\(normalize(companyCode)).\(domain)"
    }

}

File: apps/android/app/src/main/java/com/jugomo/miplancha/UsernameEmail.kt
package com.jugomo.miplancha

object UsernameEmail {

    fun normalize(valor: String) : String {
        return valor.lowercase().trim();
    }

    fun syntheticEmail(companyId: String, username: String): String {
        return "${normalize(username)}@${normalize(companyId)}.miplancha.local";
    }

}
